datafusion/sqllogictest/test_files/order.slt

```diff
@@ -1148,3 +1148,21 @@ SELECT (SELECT c from ordered_table ORDER BY c LIMIT 1) UNION ALL (SELECT 23 as
 ----
 0
 23
+
+# Do not increase partition number after fetch 1. As this will be unnecessary.
```

Comment:
This comment implies to me I wouldn't see any RepartitionExec that increases the number of partitions, but the plan has one after the `SortExec`

Maybe we could clarify the comment to better explain what is expected (maybe that the sort input should not be reparitioned)?

Reply:
Clarified the comment